src/esy.ml

```diff
@@ -91,10 +89,19 @@ module Discover = struct
          (Path.toString dir);
        [||]
      | Ok res -> res)
-    >>= fun files ->
-    files
-    |> Js.Array.map (parseFile dir)
-    |> Promise.all |> Promise.map foldResults
+    >>= fun files -> files |> Js.Array.map (parseFile dir) |> Promise.all
+
+  let getUpstreamDirs dir =
```

Comment:
This would go all the way up to root, right? I think we should stop at home. 

On Windows, `Path.split` will all result in `C:` being one of the entries in the loop, which I think is not valid too

Reply:
I don't see why we should stop at $HOME. It will make the detection work differently depending on where the user has his project. That is incredibly counter intuitive and convoluted.

Reply:
How would stopping at $HOME yield different results from the current implementation?

Files beyond HOME can be off-limits in terms of ownership

Reply:
> How would stopping at $HOME yield different results from the current implementation?

Isn't it possible for users to store their sources outside $HOME? For example, a network drive.

> Files beyond HOME can be off-limits in terms of ownership

That's the case for files inside $HOME as well. It's possible to chmod inside a home directory after all. It's a good exceptional case to point out, but we should handle it regardless.

I don't think it ever pays off to be overly clever with these things. Especially when the benefit is not clear.